hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/message/MessageFactory.java

```diff
@@ -0,0 +1,116 @@
+package pl.allegro.tech.hermes.frontend.publishing.message;
+
+import com.github.fge.jsonschema.main.JsonSchema;
+import com.google.common.collect.ImmutableMap;
+import org.apache.avro.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.common.http.MessageMetadataHeaders;
+import pl.allegro.tech.hermes.common.message.wrapper.MessageContentWrapper;
+import pl.allegro.tech.hermes.common.message.wrapper.UnsupportedContentTypeException;
+import pl.allegro.tech.hermes.domain.topic.schema.SchemaRepository;
+import pl.allegro.tech.hermes.domain.topic.schema.SchemaVersion;
+import pl.allegro.tech.hermes.domain.topic.schema.CompiledSchema;
+import pl.allegro.tech.hermes.frontend.publishing.MessageContentTypeEnforcer;
+import pl.allegro.tech.hermes.frontend.publishing.avro.AvroMessage;
+import pl.allegro.tech.hermes.frontend.publishing.metadata.HeadersPropagator;
+import pl.allegro.tech.hermes.frontend.validator.MessageValidators;
+import tech.allegro.schema.json2avro.converter.AvroConversionException;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import java.time.Clock;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+
+public class MessageFactory {
```

Comment:
I wanted to call this MessageWarrior but this would be too much awesomeness for a single class...